tools/worker-runner/provider/google/google.go

```diff
@@ -130,13 +130,10 @@ func (p *GoogleProvider) checkTerminationTime() bool {
 }
 
 func (p *GoogleProvider) WorkerStarted(state *run.State) error {
-	// start polling for graceful shutdown
-	p.terminationTicker = time.NewTicker(15 * time.Second)
 	p.proto.AddCapability("graceful-termination")
 
 	go func() {
 		for {
-			<-p.terminationTicker.C
```

Comment:
Just to be clear: removing this ticker means that we do this in an infinite loop with no wait time? (I think that's fine...just making sure I understand the code.)

Reply:
So it's a little bit confusing, I agree. The infinite for loop could _probably_ be removed, as the `p.checkTerminationTime()` call will be the one waiting for the HTTP GET response. I left the for loop in here just in case the method returned without an actual response yet from the API. I didn't look much into https://pkg.go.dev/github.com/taskcluster/httpbackoff/v3@v3.1.0#ClientDo but I thought if it had a timeout or something I'd rather just retry the `wait_for_change=true` call.

Reply:
Absolutely - yeah. I agree that keeping the loop is a good idea.

Reply:
👍🏻 yeah, mainly just a safeguard to be sure we're still listening